### PR TITLE
tests for books and categories

### PR DESCRIPTION
**Pull request by kriazantseva**
coverage

### CODE REVIEW

File: src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+        String content = result.getResponse().getContentAsString();
+        LOGGER.info("Response content: " + content);
+
+        BookDto book = objectMapper.readValue(content, BookDto.class);
+
+        assertNotNull(book);
+        assertEquals(validId, book.id());
+        assertEquals("Sherlock Holmes", book.title());
+        assertEquals("Arthur Conan Doyle", book.author());
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java]: Please save to constants and reuse it

File: src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    @WithMockUser(username = "user", roles = {"USER"})
+    public void createBook_NonAdminUser_ReturnsForbidden() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto(
+                "Another Adventures",
+                "Another Author",
+                "978-3-16-148410-0",
+                BigDecimal.valueOf(19.99),
+                "It's absolutely another story",
+                "cover.jpeg",
+                List.of(1L)
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java]: I see duplication of code, let`s create private method with building this entity

Comment (MentorMateAcademy) [reply src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java]: and feel free to use constants for values

File: src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        String requestJson = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(put("/books/{id}", 1)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestJson))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String content = result.getResponse().getContentAsString();
+        LOGGER.info("Response content: " + content);
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java]: why do you need logger in tests?

File: src/test/java/mate/academy/bookstoreapp/service/CategoryServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    @DisplayName("Delete category")
+    @Test
+    public void deleteCategory_byValidId() {
+        Long categoryId = 1L;
+
+        when(categoryRepository.findById(categoryId)).thenReturn(Optional.of(category));
+        doNothing().when(categoryRepository).deleteById(categoryId);
+        categoryService.deleteById(categoryId);
+        when(categoryRepository.findById(categoryId)).thenReturn(Optional.empty());
+
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/bookstoreapp/service/CategoryServiceTest.java]: "when" should be before action

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java (hunk 3 above)
Comment (Elena-Bruyako) [reply src/test/java/mate/academy/bookstoreapp/controller/BookControllerTest.java]: Fix comment

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
fix previous comment https://github.com/kriazantseva/book-store-app/pull/15#discussion_r1784167114

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**